sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -149,6 +149,13 @@ def test_incremental_pca_validation():
         IncrementalPCA(n_components=n_components).partial_fit(X)
 
 
+def test_n_samples_equal_n_components():
+    # Ensures no warning is raised when n_samples==n_components
+    ipca = IncrementalPCA(n_components=5)
+    ipca.partial_fit(np.random.randn(5, 7))
+    ipca.fit(np.random.randn(5, 7))
```

Comment:
it's not actually ensuring that no warning is raised.
```suggestion
    # Ensures no warning is raised when n_samples==n_components
    ipca = IncrementalPCA(n_components=5)
    with warnings.catch_warnings():
        warnings.simplefilter("error", RuntimeWarning)
        ipca.partial_fit(np.random.randn(5, 7))
    with warnings.catch_warnings():
        warnings.simplefilter("error", RuntimeWarning)
        ipca.fit(np.random.randn(5, 7))
```

Reply:
Whoops thanks